features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineView.kt

```diff
@@ -85,6 +86,24 @@ fun TimelineView(
         // TODO implement this logic once we have support to 'jump to event X' in sliding sync
     }
 
+    var lastReadMarkerIndex by rememberSaveable { mutableStateOf(Int.MAX_VALUE) }
+    var lastReadMarkerId by rememberSaveable { mutableStateOf<EventId?>(null) }
+
+    // Send a read receipt when the user has finished scrolling to an item in an index <= the last one
+    // and the event ids are different.
+    val firstVisibleIndex by remember { derivedStateOf { lazyListState.firstVisibleItemIndex } }
+    val isScrollFinished by remember { derivedStateOf { !lazyListState.isScrollInProgress } }
+    LaunchedEffect(firstVisibleIndex, isScrollFinished) {
```

Comment:
Would it make sense to put some (as much as we can) of this logic in the presenter so we can test it?

e.g.:
View would send `state.eventSink(TimelineEvents.OnScrollFinished(firstVisibleIndex))` whenever `isScrollFinished` is true.
Then you can move in the presenter all the remaining logic so it's testable.

Reply:
Done in https://github.com/vector-im/element-x-android/pull/713/commits/3f23130547704dbfe9f2db84fbb5ca7fb2766a3c.